mailmerge.py

```diff
@@ -57,13 +57,21 @@ def __init__(self, file):
                      parent.findall('{%(w)s}r/{%(w)s}fldChar[@{%(w)s}fldCharType="begin"]/..' % NAMESPACES)],
                     [children.index(e) for e in
                      parent.findall('{%(w)s}r/{%(w)s}fldChar[@{%(w)s}fldCharType="end"]/..' % NAMESPACES)],
-                    [e.text for e in
-                     parent.findall('{%(w)s}r/{%(w)s}instrText' % NAMESPACES)])
+                    [e for e in
+                     parent.findall('{%(w)s}r/{%(w)s}instrText' % NAMESPACES)]
+                )
+
                 for idx_begin, idx_end, instr in fields:
-                    m = r.match(instr)
+                    m = r.match(instr.text)
                     if m is None:
                         continue
                     parent[idx_begin] = Element('MergeField', name=m.group(1))
+
+                    # append the other tags in the w:r block too
+                    block = instr.getparent()
+                    block.remove(instr)
```

Comment:
@sliverc ok, I understood you wrong. You actually *want* to preserve the exact order.